apps/confidential/confidential.go

```diff
@@ -212,17 +217,20 @@ func WithAccessor(accessor cache.ExportReplace) Option {
 	}
 }
 
-// tokener has a shared oauth.Token object. I (jdoak) am not a fan. But at this point, that
```

Comment:
Presumably this comment was stale and meant oauth.Client.

One issue I do see though is the cache inside authorityEndpoint.  Obviously this change means there's no longer a shared cache which I'm assuming isn't great.

Reply:
This would mean every new instance of confidential client app will do the tenant discovery network call to get the endpoints and wont be able to use the cached endpoints. I think the cost is basically coming down to one network call for each instance of confidential client app to form the authorize and token endpoints in the cache. 
I think that isn't great but its not too bad either.

Reply:
Is opening an issue to track this good enough for the moment?  Assuming this doesn't actually break anything of course.

Reply:
Ok, I think issue tracking this would be good. I dont think this will break anything. Our e2e tests also make sure confidential client flows are working fine.

Reply:
Opened https://github.com/AzureAD/microsoft-authentication-library-for-go/issues/153